From e4919d717069bf59bbc374126cd157574ce261de Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 29 Jan 2023 10:03:44 +0200
Subject: [PATCH] [SW-120977] erom: remove bis chk call

Recently a patch was added in which reads bis
indication from efuse.
But efuse init was not called as part of device init
in eROM, which resulted in eROM hang.

Also, eROM is actually being erased from BIS compliant device,
so calling it from eROM is a dead code.
Hence, this API call is removed, and also efuse, and security
module are removed from linkage, as these modules are not used
anyway in eROM.

Change-Id: Ifdc5bb7882f82fe822b75682eb88cdc4e7f4783a
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 14ccba5..922db16 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -59,8 +59,6 @@
 	$(PREBOOT_ROOT_DIR)/lib/common/spi.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/spi_flash.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/lib_common.o \
-	$(PREBOOT_ROOT_DIR)/lib/common/efuse.o \
-	$(HL_DEV_ARCH)/../common/security.o \
 	$(HL_DEV_ARCH)/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/common.o \
diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 22ef76a..0296eea 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -224,3 +224,14 @@
 	ppboot_security_init();
 	erom_pcie_init();
 }
+
+/*
+ * There is no eROM for bis compliant, hence no need to check
+ * bis compliant as part of eROM. Therefore, this function is declared
+ * here as a strong function with an empty implementation, and
+ * in security module the function is declared as a weak function.
+ */
+bool preboot_check_bis_compliant_mode(void)
+{
+	return true;
+}
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 279798a..1bc2205 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1507,7 +1507,7 @@
 	reset_arc_farm_arc0();
 }
 
-bool preboot_check_bis_compliant_mode(void)
+__weak bool preboot_check_bis_compliant_mode(void)
 {
 	u32 config_fuse;
 	int ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_fuse);
